library/src/main/java/com/northconcepts/templatemaster/rest/TemplateMasterBootstrap.java

```diff
@@ -164,6 +166,12 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
         }
     }
 
+    @Override
+    public void requestInitialized(ServletRequestEvent event) {
+        HttpServletRequest request = (HttpServletRequest) event.getServletRequest();
+        RequestHolder.setHttpServletRequest(request);
```

Comment:
Aren't we already doing this in `doFilter()` for a long time?  What's changed?

BTW, we also remove the thread-local in the filter method (`RequestHolder.clearHttpServletRequest();`), do we need to do that differently as well?

Reply:
yes we are, but apparently it is not initializing it, This method ensures that it is always initialized